### PR TITLE
Upgrade to Fastify v3

### PR DESCRIPTION
**Pull request by thomheymann**
Updated types and tests for Fastify v3

#### Todo

- [x] Remove github dependencies

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Also, please update the github actions config.

File: package.json [hunk 1]
```diff
@@ -31,17 +31,18 @@
...
     "standard": "^14.0.2",
     "tap": "^12.6.6",
+    "tsd": "^0.11.0",
     "typescript": "^3.3.1"
   },
   "dependencies": {
     "cookie": "^0.4.0",
     "cookie-signature": "^1.1.0",
-    "fastify-plugin": "^1.4.0"
+    "fastify-plugin": "github:thomheymann/fastify-plugin#next"
```
Comment (jsumners) [inline package.json:45]: Why are we targeting a fork?

Comment (thomheymann) [reply package.json:45]: Because the changes required haven't been published yet

Comment (jsumners) [reply package.json:45]: Then this work should happen after a required Fastify release is available. This PR cannot be merged while it targets a fork.

Comment (thomheymann) [reply package.json:45]: Correct, Fastify v3 needs to be published before any plugin that depends on v3 can be published. 

The purpose of this piece of work is to verify that v3 typings are working correctly before publishing Fastify v3. We've already discovered multiple issues with the initial type updates by converting plugins to v3. This PR is part of that effort.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [ '12', '10', '8', '6' ]
+        node-version: [10.x, 12.x, 13.x]
```
Comment (mcollina) [inline .github/workflows/ci.yml:9]: why this change? The line after this load this value, so it should be updated as well.

Comment (thomheymann) [reply .github/workflows/ci.yml:9]: Not clear what you mean.
v6 was failing and you asked for actions config to be updated: https://github.com/fastify/fastify-cookie/pull/67#issuecomment-614158920

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [ '12', '10', '8', '6' ]
+        node-version: [10.x, 12.x, 13.x, 14.x]
```
Comment (mcollina) [inline .github/workflows/ci.yml:9]:
```suggestion
        node-version: [10.x, 12.x, 14.x]
```

File: package.json [hunk 4]
```diff
@@ -31,21 +31,20 @@
   "homepage": "https://github.com/fastify/fastify-cookie#readme",
   "devDependencies": {
     "@types/node": "^13.1.0",
-    "fastify": "^2.11.0",
+    "fastify": "^3.0.0-alpha.1",
```
Comment (mcollina) [inline package.json:34]:
```suggestion
    "fastify": "^3.0.0-rc.1",
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [ '12', '10', '8', '6' ]
+        node-version: [10.x, 12.x, 14.x]
```
Comment (mcollina) [inline .github/workflows/ci.yml:9]:
```suggestion
        node: [10.x, 12.x, 14.x]
```
